demos/form-control/dropdown-plus.php

```diff
@@ -21,7 +21,7 @@
 
 $form->addControl('category_id', [Form\Control\Dropdown::class, 'model' => new Category($app->db)]);
 $form->addControl('sub_category_id', [Form\Control\DropdownCascade::class, 'cascadeFrom' => 'category_id', 'reference' => Category::hinting()->fieldName()->SubCategories]);
-// DEBUG only, uncomment before merge!!! $form->addControl('product_id', [Form\Control\DropdownCascade::class, 'cascadeFrom' => 'sub_category_id', 'reference' => SubCategory::hinting()->fieldName()->Products]);
+$form->addControl('product_id', [Form\Control\DropdownCascade::class, 'cascadeFrom' => 'sub_category_id', 'reference' => SubCategory::hinting()->fieldName()->Products]);
```

Comment:
when this line is commented, tests pass

@georgehristov can you please help with fixing this?

Reply:
The issue is caused by automatic creation of `their_field` value:

https://github.com/atk4/data/blob/07803e48828a2b6c12b9c05cb369344e5fd929ca/src/Reference/HasMany.php#L58

In this case it defaults to `product_category_id` and used in scope condition but field is not found due to your renaming.

Reply:
Thanks, I created a PR for it https://github.com/atk4/data/pull/830, wdyt?

Reply:
Thanks, I created a PR for it https://github.com/atk4/data/pull/830 . We should not guess any references at runtime, if absolutely needed, then lately on __construct